tests/libres_tests/res/analysis/test_ies.py

```diff
@@ -0,0 +1,118 @@
+#  Copyright (C) 2013  Equinor ASA, Norway.
+#
+#  The file 'test_analysis_module.py' is part of ERT - Ensemble based Reservoir Tool.
+#
+#  ERT is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+#  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+#  FITNESS FOR A PARTICULAR PURPOSE.
+#
+#  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+#  for more details.
+
+import random
+import sys
+
+from ecl.util.enums import RngAlgTypeEnum, RngInitModeEnum
+from ecl.util.util import BoolVector, RandomNumberGenerator
+from libres_utils import ResTest, tmpdir
+
+from res.analysis import AnalysisModule, AnalysisModuleLoadStatusEnum
+from res.enkf import MeasData, ObsData
+from res.util import Matrix
+
+
+def forward_model(params, model_error=False):
+    # Y = A*p[0] + B*p[1]
+    A = 2
+    B = -1
+    C = 1
+    D = 0.1
+    state = [A * params[0] + B * params[1], C * params[0] - D * params[1] * params[0]]
+    return state
+
+
+def measure(state):
+    return 0.25 * state[0] - 0.1 * state[1] * state[1]
+
+
+def init_matrices(ens, mask, obs, rng):
+    state_size = 2
+    report_step = 5
+    meas_data = MeasData(mask)
+    meas_block = meas_data.addBlock("OBS", report_step, len(obs))
+
+    A = Matrix(state_size, mask.countEqual(True))
+    active_iens = 0
+    for iens, params in enumerate(ens):
+        if mask[iens]:
+            state = forward_model(params)
+            meas_block[0, iens] = measure(state)
+
+            A[0, active_iens] = params[0]
+            A[1, active_iens] = params[1]
+
+            active_iens += 1
+
+    S = meas_data.createS()
+
+    obs_data = ObsData()
+    obs_block = obs_data.addBlock("OBS", 1)
+    for iobs, obs_value in enumerate(obs):
+        obs_block[iobs] = obs_value
+
+    R = obs_data.createR()
+    dObs = obs_data.createDObs()
+    E = obs_data.createE(rng, meas_data.getActiveEnsSize())
+    D = obs_data.createD(E, S)
+
+    obs_data.scale(S, E=E, D=D, R=R, D_obs=dObs)
+    return (A, S, E, D, R, dObs)
+
+
+def create_analysis_module(lib_name):
+    return AnalysisModule(lib_name=f"{lib_name}.so")
+
+
+class IESTest(ResTest):
```

Comment:
There is a lot of reuse of code between this test and the RML test. Would it be possible to put these together, or are there subtle differences between the tests that I don't see from a first glance?

Reply:
That is possible but I decided against it because the RML test will be gone once https://github.com/equinor/ert/pull/2037 is merged